### ISSUE TITLE
sampler not working

### ISSUE THREAD

**Issue by ShaheenPerveen** [cfotache/pytorch_imageclassifier#2]
Hi,

I tried using your code for creating train and test data loader from one folder. The code seems perfect to me but somehow its not working in my case.
I have an image folder with 4 sub-directories corresponding to each class. When I am running your code, the size of trainloader and testloader is same i,e, full-size of the image folder. Could you please help me understand what am I doing wrong?
 This is the code snippet:

def load_split_train_test(datadir, test_size = .2):
    train_transforms = transforms.Compose([transforms.Resize(256),
                                       transforms.CenterCrop(224),
                                       transforms.ToTensor(),
                                       transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])])
    
    test_transforms = transforms.Compose([transforms.Resize(256),
                                       transforms.CenterCrop(224),
                                       transforms.ToTensor(),
                                       transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])])
    
    train_data = datasets.ImageFolder(datadir, transform=train_transforms)
    
    test_data = datasets.ImageFolder(datadir, transform=test_transforms)
    
    num_train = len(train_data)
    indices = list(range(num_train))
    split = int(np.floor(test_size * num_train))
    np.random.shuffle(indices)
    
    train_idx, test_idx = indices[split:], indices[:split]
    
    train_sampler = SubsetRandomSampler(train_idx)
    test_sampler = SubsetRandomSampler(test_idx)
    
    trainloader = torch.utils.data.DataLoader(train_data, sampler=train_sampler, batch_size=batch_size)
    testloader = torch.utils.data.DataLoader(test_data, sampler=test_sampler, batch_size=batch_size)

    return trainloader, testloader
